core/templates/pages/exploration-editor-page/services/learner-answer-details-data.service.ts

```diff
@@ -17,81 +17,95 @@
+        entity_id : this._expId
+        });
+        return this.http.get(learnerAnswerInfoDataUrl).toPromise();
+    }
+    private _deleteLearnerAnswerInfo(entityId, stateName, learnerAnswerInfoId){
+      const learnerAnswerInfoDataUrl = this.urlInterpolationService.interpolateUrl(
+        this.LEARNER_ANSWER_INFO_DATA_URL,{
+          entity_type:'exploration',
+          entity_id:entityId
+        });
+      return this.http.delete(learnerAnswerInfoDataUrl, {
         params: {
           state_name: stateName,
           learner_answer_info_id: learnerAnswerInfoId
         }
       });
     };
-
-    return {
-      getData: function() {
-        return _data;
-      },
-      fetchLearnerAnswerInfoData: function() {
-        return _fetchLearnerAnswerInfoData().then(function(response) {
-          learnerAnswerInfoData =
-            response.data.learner_answer_info_data;
-          for (var i = 0; i < learnerAnswerInfoData.length; i++) {
-            var stateName = learnerAnswerInfoData[i].state_name;
-            var interactionId = learnerAnswerInfoData[i].interaction_id;
-            var customizationArgs = learnerAnswerInfoData[i].customization_args;
-            var learnerAnswerInfoDicts = (
-              learnerAnswerInfoData[i].learner_answer_info_dicts);
-            var learnerAnswerDetails = (
-              // eslint-disable-next-line max-len
-              LearnerAnswerDetailsObjectFactory.createDefaultLearnerAnswerDetails(
-                _expId, stateName, interactionId, customizationArgs,
+    public getData() {
+      return this._data;
+    }
+    public fetchLearnerAnswerInfoData() {
+      return this._fetchLearnerAnswerInfoData().toPromise().then(
+        (response)=>{
+          this.learnerAnswerInfoData = response.body.learner_answer_info_data;
+          for (let i = 0;i < this.learnerAnswerInfoData.length;i++) {
+            const stateName = this.learnerAnswerInfoData[i].state_name;
+            const interactionId = this.learnerAnswerInfoData[i].interaction_id;
+            const customizationArgs = 
+            this.learnerAnswerInfoData[i].customization_args;
+            const learnerAnswerInfoDicts = (
+              this.learnerAnswerInfoData[i].learner_answer_info_dicts
+            );
+            const learnerAnswerDetails = (
+              LearnerAnswerDetails.createDefaultLearnerAnswerDetails(
+                this._expId, stateName, interactionId, customizationArgs,
                 learnerAnswerInfoDicts.map(
-                  LearnerAnswerInfoObjectFactory.createFromBackendDict)));
-            _data.push(learnerAnswerDetails);
+                  LearnerAnswerInfo.createFormBackendDict
+                )
+              )
+            );
+            this._data.push(learnerAnswerDetails);
           }
-          return response.data;
-        });
+          return response.body;
+        }
+      );
+    }
+    public deleteLearnerAnswerInfo(entityId,stateName, learnerAnswerInfoId) {
```

Comment:
Add space after comma in comma-separated values like in function arguments
```suggestion
    public deleteLearnerAnswerInfo(entityId, stateName, learnerAnswerInfoId) {
```

Reply:
Done